From dce2f990d7dea49dafc2818dc8a5d6d0fc2ae82b Mon Sep 17 00:00:00 2001
From: shengfei Xu <xsf@rock-chips.com>
Date: Mon, 23 Oct 2017 18:00:34 +0800
Subject: [PATCH] regulator: rk808: fix the regulator control value in system
 suspend for rk816

set 1 if the regulator is enabled in system suspend, 0 if not.

Change-Id: I8d51ac685bbd2417f440842d010fe47946c9f567
Signed-off-by: shengfei Xu <xsf@rock-chips.com>
---
 drivers/regulator/rk808-regulator.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/regulator/rk808-regulator.c b/drivers/regulator/rk808-regulator.c
index d827c4165e54..3218c77136f9 100644
--- a/drivers/regulator/rk808-regulator.c
+++ b/drivers/regulator/rk808-regulator.c
@@ -370,7 +370,7 @@ static int rk816_set_suspend_enable(struct regulator_dev *rdev)
 
 	return regmap_update_bits(rdev->regmap, reg,
 				  val,
-				  0);
+				  val);
 }
 
 static int rk816_set_suspend_disable(struct regulator_dev *rdev)
@@ -400,7 +400,7 @@ static int rk816_set_suspend_disable(struct regulator_dev *rdev)
 
 	return regmap_update_bits(rdev->regmap, reg,
 				  val,
-				  val);
+				  0);
 }
 
 static int rk808_set_suspend_enable(struct regulator_dev *rdev)
-- 
2.35.3

